pygsheets/authorization.py

```diff
@@ -0,0 +1,69 @@
+import os
+import json
+
+from google.oauth2 import service_account
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import Flow
+
+from pygsheets.client import Client
+
+
+def _get_user_authentication_credentials(client_secret_file, scopes, credential_directory=None):
+    """"""
+    if credential_directory is None:
+        credential_directory = os.getcwd()
+    elif credential_directory == 'global':
+        home_dir = os.path.expanduser('~')
+        credential_directory = os.path.join(home_dir, '.credentials')
+        if not os.path.exists(credential_directory):
+            os.makedirs(credential_directory)
+    else:
+        pass
+
+    credentials_path = os.path.join(credential_directory, 'sheets.googleapis.com-python.json')
+
+    if os.path.exists(credentials_path):
+        # expect these to be valid. may expire at some point, but should be refreshed by google api client...
+        return Credentials.from_authorized_user_file(credentials_path, scopes=scopes)
+
+    flow = Flow.from_client_secrets_file(client_secret_file, scopes=scopes, redirect_uri='urn:ietf:wg:oauth:2.0:oob')
+
+    auth_url, _ = flow.authorization_url(prompt='consent')
+
+    print('Please go to this URL and finish the authentication flow: {}'.format(auth_url))
+    code = input('Enter the authorization code: ')
+    flow.fetch_token(code=code)
+
+    credentials = flow.credentials
+
+    credentials_as_dict = {
+        'token': credentials.token,
+        'refresh_token': credentials.refresh_token,
+        'id_token': credentials.id_token,
+        'token_uri': credentials.token_uri,
+        'client_id': credentials.client_id,
+        'client_secret': credentials.client_secret
+    }
+
+    with open(credentials_path, 'w') as file:
+        file.write(json.dumps(credentials_as_dict))
+
+    return credentials
+
+
+def authorize(client_secret='client_secret.json', service_account_file=None, credentials_directory='',
```

Comment:
Please change the param names to match with the old convention, this will break old codes. Is there any specific reason to change the param names? Also please add an auth-non local param. and use the docs from previous version you are missing some imp info like `credentials_directory` can be global etc

Reply:
I changed it because I find abbreviated parameter names are often unclear what they are for. And I though it would be okay for people to change this, as the most common usage with client_secret.json would need not change.

But if you disagree I will change it.

And will improve the docs.

Reply:
changing names is fine, but make sure the code works with older names too, with a deprecation warning maybe.

Reply:
The reason I removed the non_local option is that, the new library only supports non_local_options, by default and I can't get the browser option to work. 

To open a browser it wants to instantiate a web server, which makes the auth-flow way slower.

So non_local is just the default.

Reply:
Oh i see. cool then